src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -304,4 +305,25 @@ private static function computeRegexp(array $tokens, int $index, int $firstOptio
             }
         }
     }
+
+    private static function removeCapturingGroups(string $regexp)
```

Comment:
The name is irritating. It does not remove the groups but transforms capturing to non-capturing groups.

Reply:
what's irritating is finding another name :)

Reply:
how about `transformCapturingGroups` or `transformCapturingGroupsToNonCapturing`

Reply:
done